[dylkaw] iP

dylkaw opened the pull request:
# Glub

> "The best task manager I've ever used!" - A satisfied user

Glub frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is:
1. download it from [here](https://github.com/dylkaw/ip/releases)
2. double click it
3. add your tasks
4. let it manage your tasks for you :wink:

and it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
***
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        Glub glub = new Glub(new Storage("tasks.txt"));
        Ui.greet();
        glub.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

dylkaw committed:
renaming bot and creating greeting

dylkaw committed:
reducing greeting message

dylkaw committed:
adding echo and exit functions

dylkaw committed:
creating task class

dylkaw committed:
creating addTask function

dylkaw committed:
adding list function

dylkaw committed:
adding getDoneIcon, toggleDone methods to Task class

dylkaw committed:
changing toggleDone to 2 separate methods

dylkaw committed:
adding mark function

dylkaw committed:
adding unmark method

dylkaw committed:
adding deadline, event, todo classes

dylkaw committed:
improving addTask method to be able to add different tasks

dylkaw committed:
adding exceptions to handle missing inputs or unknown inputs

dylkaw committed:
adding deleteTask method

dylkaw committed:
reducing exceptions into single GlubException class

dylkaw committed:
creating automated text tests

dylkaw committed:
Add isListening variable

dylkaw committed:
Add taskType enum

dylkaw committed:
List imported classes explicitly

dylkaw committed:
Add save and load

Classes have been refactored for loading

dylkaw committed:
Add date handling for deadline and event tasks

dylkaw committed:
Extract Parser, Storage, TaskList and UI classes

dylkaw committed:
Rename Duke to Glub

dylkaw committed:
Create glub and glub.task packages

dylkaw committed:
Set up gradle

dylkaw committed:
Merge branch 'add-gradle-support'

dylkaw committed:
Update build.gradle file for Glub

dylkaw committed:
Add TaskList and Todo tests

dylkaw committed:
Add empty storage file for tests

dylkaw committed:
Add JAR creation to build.gradle

dylkaw committed:
Add JavaDocs

dylkaw committed:
Adjust case indentation to coding standard

dylkaw committed:
Add find command

dylkaw committed:
Merge branch 'branch-A-CodingStandard'

dylkaw committed:
Merge branch 'branch-Level-9'

dylkaw committed:
Add checkstyle

dylkaw committed:
Fix some style violations

TohLiYuan commented on src/main/java/glub/task/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+            throw new GlubException("Oops! Task not recognised!");
+        }
+        storage.saveTasks(this.taskList);
+    }
```
Should these be handled in the Parser instead of in the Task class?

TohLiYuan commented on src/main/java/glub/task/TaskType.java:
```diff
@@ -0,0 +1,7 @@
+    TODO,
+    DEADLINE,
+    EVENT
+}
```
Good use of Enums

TohLiYuan reviewed (commented):
Good overall, could do with some minor optimisations but not strictly necessary

qz1004 commented on src/main/java/glub/task/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+
+import java.time.LocalDateTime;
+
+public class Deadline extends Task{
```
Remember to add a space before the {

qz1004 commented on src/main/java/glub/task/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+    public void addTask(String task, TaskType taskType, boolean isDone) throws GlubException {
+        DateTimeFormatter dateTimeFormat = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+        if (task.equals("")) {
+            throw new GlubException(String.format("OOPS!! The description of a %s cannot be empty.\n", taskType));
```
Should this be handled in the Parser class?

qz1004 reviewed (commented):
Looks good! I found your code easy to read with good adherence to coding standards. I like your use of case and enums. You can consider moving some of the string parsing into the Parser class.

dylkaw committed:
Implement GUI

dylkaw committed:
Add assertions to Parser, Storage and TaskList

Directory path in Storage, input in  parser and task type in addTask
method in TaskList may be null.

Null variables will cause the program to crash.

Assertions are added to check if these variables are null and exit.

These assumptions will allow us to detect programming errors.

dylkaw committed:
Edit error message for addTask

Original error message tells the user that the correct date format is
dd/MM/yyyy HH:mm but the correct one actually is dd-MM-yyyy HHmm.

This has been corrected.

dylkaw committed:
Remove printHorizontalLine method

printHorizontalLine is not necessary after moving to a GUI from CLI.

dylkaw committed:
Improve code quality

Some methods were too long, so fragments were abstracted into methods.

Code style issues were also addressed by adding javadocs.

dylkaw committed:
Merge pull request #1 from dylkaw/branch-A-Assertions

Add assertions to Parser, Storage and TaskList

dylkaw committed:
Merge branch 'master' into branch-A-CodeQuality

dylkaw committed:
Merge pull request #2 from dylkaw/branch-A-CodeQuality

Improve code quality

dylkaw committed:
Fix bug in addTask method

addTask method asked for a TaskType for taskType argument when it
should be type String.

dylkaw committed:
Add tagging to tasks

dylkaw committed:
Add support for saving tags

dylkaw committed:
Fix issue where inputting a deadline fails

dylkaw committed:
Fix checkstyle violations

dylkaw committed:
Ensure javadocs follow coding standard

dylkaw committed:
Add project name to title bar of window

dylkaw committed:
Add welcome message

dylkaw committed:
Update user guide

dylkaw committed:
Update unknown error message to show all commands

dylkaw committed:
Update tests for task list and todo
